context/datapath.go

```diff
@@ -154,23 +153,6 @@ func (node *DataPathNode) ActivateUpLinkTunnel(smContext *SMContext) error {
 		return err
 	}
 
-	var teid uint32
-	var teidErr error
-
-	if factory.SmfConfig.Configuration.EnableDbStore {
-		var tmp int32
-		tmp, teidErr = smfContext.DrsmCtxts.TeidPool.AllocateInt32ID()
-		teid = uint32(tmp)
-	} else {
-		teid, teidErr = destUPF.GenerateTEID()
-	}
-	if teidErr != nil {
-		logger.CtxLog.Errorf("generate uplink TEID fail: %s", teidErr)
-		return teidErr
-	} else {
-		node.UpLinkTunnel.TEID = teid
```

Comment:
Should we not keep teid in the data structure as is. Irrespective of who allocates TEID?

Reply:
The data structure for the GTPTunnel and its TEID hasn't changed, the TEID is simply set at a different moment. Now it is set when we receive the PFCP session establishment response based on the response content from the UPF.